### ISSUE TITLE
Implicit fall-through error in arch/x86/kvm/cpuid.c

### ISSUE THREAD

**Issue by isanbard** [ClangBuiltLinux/linux#1616]
I'm seeing this error with upstream. The default statement consists of this:

```
    default:
        break;
```

Should Clang produce a diagnostic in this situation?

```
[arch/x86/kvm/cpuid.c:739](https://cs.corp.google.com/piper///depot/google3/arch/x86/kvm/cpuid.c?l=739):2: error: unannotated fall-through between switch labels [-Werror,-Wimplicit-fallthrough]
        default:
        ^
[arch/x86/kvm/cpuid.c:739](https://cs.corp.google.com/piper///depot/google3/arch/x86/kvm/cpuid.c?l=739):2: note: insert 'break;' to avoid fall-through
        default:
        ^
        break; 
1 error generated.
```

**Comment by nathanchance** [comment]
See #636 for some more context around Clang vs. GCC in that area. TL;DR: We believe Clang's version of `-Wimplicit-fallthrough` is more correct than GCC's, so we would not like to match its semantics.

I submitted https://lore.kernel.org/r/20220322152906.112164-1-nathan@kernel.org/ but I forgot to open an issue and it seems like it is not in -next yet...

**Comment by isanbard** [comment]
Thanks, @nathanchance. Yeah, I could go either way on whether clang should match gcc with this.

**Comment by nathanchance** [comment]
Merged into mainline (5.18-rc1): https://git.kernel.org/linus/07ea4ab1f9b83953ff5c3f6ccfb84d581bfe0046
